src/Utility/HDF5Writer.C

```diff
@@ -349,6 +349,8 @@ void HDF5Writer::writeSIM (int level, const DataEntry& entry,
   {
     Vector pchVec;
     size_t ncmp = sim->getProblem()->getNoFields(2);
+    if (ncmp == 0)
+      return;
```

Comment:
This leaves the temporary `pchVec` unused if `ncmp == 0`, so maybe move its definition after the if-statement

Reply:
done.